dvc/remote/ssh.py

```diff
@@ -44,8 +46,9 @@ class RemoteSSHCmdError(RemoteBaseCmdError):
 class RemoteSSH(RemoteBase):
     scheme = 'ssh'
 
-    # NOTE: temporarily only absolute paths are allowed
-    REGEX = r'^ssh://((?P<user>.*)@)?(?P<host>[^/]*):(?P<path>/.*)$'
+    # NOTE: we support both URL-like (ssh://[user@]host.xz[:port]/path) and
+    # SCP-like (ssh://[user@]host.xz:path) urls.
```

Comment:
mhm, @efiop , this is from the `scp` manpage:

> The source and target may be specified as a local pathname, a remote host
> with optional path in the form [user@]host:[path], or a URI in the form
>  `scp://[user@]host[:port][/path].`  Local file names can be made explicit using absolute
> or relative pathnames to avoid scp treating file names containing ‘:’ as host specifiers.

I don't think we need to support other than the URI form for both (scp or ssh), in this case, we have two valid schemas, similar to the RemoteHTTP (https and htttp), maybe, instead of using a regex, it would be easier to specify the schema in `valid_schemas=['http', 'https']` and use `if parsed.schema in valid_schemas` or something like that

Reply:
I didn't notice it on the first review :disappointed:

Reply:
urlparse is able to parse both, so as long as that works we should be fine. What we really care about is that url should start with `ssh://`. If I recall correctly, we did use urlparse for validation in the past, but since we had some weird rules(e.g. see azure with weird stuff like `azure://Containername=test;Connectionstring=abcd;` which luckily now obsoleted) I've moved to using custom regexes. There are almost no wierd rules right now, so I think we can now consider moving back to urlparse once again. Created https://github.com/iterative/dvc/issues/1477 .